plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/session/CodeGenerationState.kt

```diff
@@ -100,6 +109,7 @@ class CodeGenerationState(
                 amazonqRepositorySize = repositorySize,
                 result = result,
                 reason = failureReason,
+                reasonDesc = failureReasonDesc,
```

Comment:
I see we pass `operation` around, but not seeing where we emit it.

Reply:
It's exposed inside `FeatureDevException#reasonDesc()`. Rather than have the instrumentation try to build `reasonDesc` consistently, `FeatureDevException` exposes how to get it, including both the operation and additional desc (if present).

```
fun reasonDesc(): String {
        return when (desc) {
            desc -> String.format("%s: %s", this.operation, desc) // <-- Here
            else -> String.format("%s", this.operation)
        }
}
```